internal/api/courses/admin/email.go

```diff
@@ -0,0 +1,66 @@
+package admin
+
+import (
+	"errors"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/email"
+)
+
+type EmailRequest struct {
+	core.APIRequestCourseUserContext
+	core.MinCourseRoleGrader
+
+	email.Message
+
+	DryRun bool `json:"dry-run"`
+}
+
+type EmailResponse struct {
+	To  []string `json:"to"`
+	CC  []string `json:"cc"`
+	BCC []string `json:"bcc"`
+}
+
+// Send an email to course users.
+func HandleEmail(request *EmailRequest) (*EmailResponse, *core.APIError) {
+	response := EmailResponse{}
+	var err error
+
+	if request.Subject == "" {
+		return nil, core.NewBadRequestError("-627", &request.APIRequest, "No email subject provided.")
+	}
+
+	var errs error
+
+	request.To, err = db.ResolveCourseUsers(request.Course, request.To)
+	errs = errors.Join(errs, err)
+
+	request.CC, err = db.ResolveCourseUsers(request.Course, request.CC)
+	errs = errors.Join(errs, err)
+
+	request.BCC, err = db.ResolveCourseUsers(request.Course, request.BCC)
+	errs = errors.Join(errs, err)
+
+	if errs != nil {
+		return nil, core.NewInternalError("-628", &request.APIRequestCourseUserContext, "Failed to resolve course users.").Err(errs)
```

Comment:
This error message is a bit off.
An end user does not know what you mean by "course users".
You can say something they would understand like "email recipients".